.github/workflows/eks-ci.yml

```diff
@@ -0,0 +1,134 @@
+name: EKS-CI
+
+on:
+  # On demand execution of workflow will run all suites.
+  workflow_dispatch:
+    inputs:
+      test-targets:
+        required: true
+        description: "Specify test markers to run"
+        default: '[\"pre_merge\"]'
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  GITHUB_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  CONTAINER_SUFFIX: ${{ github.run_id }}
+  TEST_TARGETS_DEFAULT: '[\"pre_merge\"]'
+  AWS_REGION: eu-west-2
+  REPORTS_DIR: tests/allure/results/
+  KUBE_NAMESPACE: kube-system
+
+permissions:
+  actions: read
+  checks: write
+  statuses: read
+
+jobs:
+  Setup:
+    runs-on: ubuntu-20.04
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    timeout-minutes: 15
+    steps:
+    - id: Default
+      env:
+        TEST_TARGETS: ${{ env.TEST_TARGETS_DEFAULT }}
+      run: |
+        echo "TEST_MATRIX=$TEST_TARGETS" >> $GITHUB_ENV
+
+    - id: Dispatch
+      if: github.event_name == 'workflow_dispatch'
+      env:
+        TEST_TARGETS: ${{ github.event.inputs.test-targets }}
+      run: echo "TEST_MATRIX=${TEST_TARGETS}" >> $GITHUB_ENV
+
+    - id: set-matrix
+      run: echo "::set-output name=matrix::${{ env.TEST_MATRIX}}"
```

Comment:
set-output is [deprecated](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands)